### ISSUE TITLE
Port registry React/NextJS code to Registry-V2 Repo

### ISSUE THREAD

**Issue by bcpeinhardt** [coder/internal#122]
The registry-v2 project is currently set up with a default react app embedded into the binary. We need to port the existing react code in registry.coder.com into registry-v2. Once we have parity in functionality, we can start looking at SEO improvements/static generation, but we probably want to first focus on the necessary work to get the newer registry deployed first.

**Comment by matifali** [comment]
Will the new registry get built and redeployed on each tag pushed to `coder/modules`?

**Comment by bcpeinhardt** [comment]
The idea right now is every merge to main.

**Comment by buenos-nachos** [comment]
This is something we'll worry about down the line, but one thing Ben and I talked about is that the repo is small enough that it would be easy to switch the styling logic over to Tailwind

The core porting obviously comes first, though

**Comment by matifali** [comment]
The building on merges to main is safer as it will be always ahead of individual tags. (Given we push tags to main).

**Comment by buenos-nachos** [comment]
Making a quick update:
1. I think I'm getting pretty close to wrapping up all the frontend code. Took a little bit of time to get the foundation in place, and figure out what had to be rewritten, but now that that's figured out, I'm zipping through things pretty quickly
2. The main problem I've been running into so far (and to be clear, this is something I'm mentioning for the first time – any communication problems are solely on me) is that we're treating the READMEs for each module and template as a source of truth for several values. I put mock data in place to get around that and make sure I didn't get blocked, but we need to get some kind of parsing setup figured out
   1. Honestly, I don't think it's a huge priority until I'm back from vacation and can start on the frontend again, so I wouldn't mind having this be by first foray into writing Go. All the README metadata is contained in frontmatter, so it shouldn't be that hard to parse it out.
   2. I also think that as a general setup, the frontend should be as lean as possible, and if we can move parsing logic to the backend, that'll help with things like page load times
   3. If @bcpeinhardt wants to take this while I'm gone, I'm totally cool with that, though
3. I've got the basics of the details/slug pages in place. The good news is that both pages have pretty close parity, so when I finish one, the other will be close to done, too
4. When I get back, I'll make a sub-issue for doing a quick round of polish once I wrap up the broad strokes of everything else

**Comment by buenos-nachos** [comment]
At this point, I would say that pretty much all the React functionality has been converted over, and it's mostly the Next.js server logic that is still a work in progress. Going to start helping out with some of the Go rewrites.
